File: src/main/java/com/team10/backend/global/security/JwtAuthenticationFilter.java
package com.team10.backend.global.security;

import com.team10.backend.global.util.CookieUtil;
import io.jsonwebtoken.ExpiredJwtException;
import io.jsonwebtoken.JwtException;
import jakarta.servlet.FilterChain;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;
import org.springframework.util.StringUtils;
import org.springframework.web.filter.OncePerRequestFilter;

import java.io.IOException;

import static com.team10.backend.global.constant.CookieConstants.ACCESS_TOKEN;

//@Slf4j
@Component
public class JwtAuthenticationFilter extends OncePerRequestFilter {

    private final CookieUtil cookieUtil;
    private final TokenProvider tokenProvider;

    @Override
    protected void doFilterInternal(HttpServletRequest request,
                                    HttpServletResponse response,
                                    FilterChain filterChain) throws ServletException, IOException {
        // 쿠키 확인
        String token = cookieUtil.getCookieValue(request, ACCESS_TOKEN);

        if (!StringUtils.hasText(token)) {
            filterChain.doFilter(request, response);
            return;
        }

        try {
            Authentication authentication = tokenProvider.getAuthentication(token);
            SecurityContextHolder.getContext().setAuthentication(authentication);
        } catch (ExpiredJwtException e) {
//            log.debug("JWT expired");
        } catch (JwtException e) {
//            log.warn("Invalid JWT");
        } catch (Exception e) {
//            log.error("Unexpected error in JWT filter", e);
        }

        filterChain.doFilter(request, response);
    }

    public JwtAuthenticationFilter(CookieUtil cookieUtil, TokenProvider tokenProvider) {
        this.cookieUtil = cookieUtil;
        this.tokenProvider = tokenProvider;
    }
}

File: src/main/kotlin/com/team10/backend/global/security/JwtAuthenticationFilter.kt
package com.team10.backend.global.security

import com.team10.backend.global.constant.CookieConstants.ACCESS_TOKEN
import com.team10.backend.global.util.CookieUtil
import io.jsonwebtoken.ExpiredJwtException
import io.jsonwebtoken.JwtException
import jakarta.servlet.FilterChain
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.slf4j.LoggerFactory
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component
import org.springframework.web.filter.OncePerRequestFilter

@Component
class JwtAuthenticationFilter(
    private val cookieUtil: CookieUtil,
    private val tokenProvider: TokenProvider
) : OncePerRequestFilter() {

    private val log = LoggerFactory.getLogger(javaClass)

    override fun doFilterInternal(
        request: HttpServletRequest,
        response: HttpServletResponse,
        filterChain: FilterChain
    ) {
        val token = cookieUtil.getCookieValue(request, ACCESS_TOKEN)

        if (token.isNullOrBlank()) {
            filterChain.doFilter(request, response)
            return
        }

        try {
            val authentication = tokenProvider.getAuthentication(token)
            SecurityContextHolder.getContext().authentication = authentication
        } catch (e: ExpiredJwtException) {
            log.debug("JWT expired")
        } catch (e: JwtException) {
            log.warn("Invalid JWT")
        } catch (e: Exception) {
            log.error("Unexpected error in JWT filter", e)
        }

        filterChain.doFilter(request, response)
    }
}
